src/main/java/com/fleotadezuta/youthprogrammanager/service/ChildService.java

```diff
@@ -39,4 +39,14 @@ public Mono<ChildDto> deleteChild(String id) {
                         .then(Mono.just(child)))
                 .map(childMapper::fromChildDocumentToChildDto);
     }
+
+    public Mono<ChildDto> updateChild(String id, ChildDto childDto) {
+        var childDoc = childMapper.fromChildDtoToChildDocument(childDto);
+        return childRepository.findById(id)
+                .flatMap(existingChildDoc -> {
+                    childDoc.setId(existingChildDoc.getId());
+                    return childRepository.save(childDoc);
+                })
+                .map(childMapper::fromChildDocumentToChildDto);
+    }
```

Comment:
Itt szerintem felesleges a findBy, megvan az ID és a document többi része is (a DTO-ban).
Illetve még szebb lenne, ha az egész egy hívási lánc lenne, valahogy így:
```java
return Mono.just(childDto)
                .map(childMapper::fromChildDtoToChildDocument)
                .flatMap(existingChildDoc -> {
                    childDoc.setId(existingChildDoc.getId());
                    return childRepository.save(childDoc);
                })
                .map(childMapper::fromChildDocumentToChildDto);
```